server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -339,7 +341,14 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
      * @return the resolved path
      */
     public static Path resolveNodePath(final Path path, final int nodeLockId) {
-        return path.resolve(NODES_FOLDER).resolve(Integer.toString(nodeLockId));
```

Comment:
This is a modification which differs from the original pr to make sure subfolder are not recreated twice.